### ISSUE TITLE
DNG from Huawei phone imported into Saulala produces weird result

### ISSUE THREAD

**Issue by EaryChow** [StudioPetrikas/Saulala#11]
[IMG_20230512_181009.dng.txt](https://github.com/StudioPetrikas/Saulala/files/13991871/IMG_20230512_181009.dng.txt)

Delete the .txt at the end. I have to do this since GitHub doesn't support uploading DNG files.

This is what it looks like in the APP:
![655c3658a289647cb173586500fc29d7](https://github.com/StudioPetrikas/Saulala/assets/59176246/762b0c4a-c97f-4996-bee0-3bdc113bd41a)

And what it looks like in Darktable:
![Darktable_HuaweiDNG](https://github.com/StudioPetrikas/Saulala/assets/59176246/509a75a3-6942-47db-bb3b-c65d8991550a)

BTW. Can you provide an APK installation file on the official website? Huawei phones are banned from installing Google Play (Chinese people don't use Google Play at all even before this but anyways), I have a Samsung phone with Google Play in it but that phone doesn't produce DNG file. So I ultimately have to use Huawei Phone's Camera to produce the DNG file, bring it to Samsung phone for Saulala. I might as well just use Darktable if I have to go through the transmission between devices.

**Comment by StudioPetrikas** [comment]
So there are a few reason why these differ:

1. Saulala does not have GainMap implemented. (In Darktable, go to "raw black/white point" module and select "none" for flat field correction.) This is something that I cannot implement due to lack of knowledge.
2. Darktable boosts exposure by default. Disable that module.
3. Darktable uses 'colour calibration', Saulala is using Camera White Balance. So disable "colour calibration" module and in the "white balance" module, select "As Shot" (click on the camera button)
4. "Filmic" also applies a curve that might be different from the one in AgX. 

If you've done all that, you should be getting a pretty similar result:
![screens](https://github.com/StudioPetrikas/Saulala/assets/70055734/44ae987c-b926-4d14-9347-74c90b0e9019)

Lastly, I would really love to avoid posting the .apk on the website. Google provides good analytics, automatic updates, ability to take down or roll back applications that are already installed. I would love to retain that control. 

I'm sure websites exist that index cache these APKs straight off the Google Store, find the one you feel is trustworthy and use the apk from there. (Hopefully they update it as soon as I push an update to the store). Sorry about that.

**Comment by EaryChow** [comment]
I followed your step and it still seems completely different, Saulala gives me almost a Log encoding kind of less chroma laden result.
![image](https://github.com/StudioPetrikas/Saulala/assets/59176246/dd015f5d-313b-4713-999f-2ce78ae25cc2)

I am using Windoes Phone Link to put them on the same screen for the screenshot.

**Comment by StudioPetrikas** [comment]
Is "Sauce" set to AgX?
Have you tried exporting and comparing the output?

Since I'm getting a "close enough" match on my end, I'd be interested to troubleshoot what is causing the issue on yours.

**Comment by EaryChow** [comment]
> Is "Sauce" set to AgX?

Yes

> Have you tried exporting and comparing the output?

Here is exported from Saulala:
![IMG_20230512_181009_Sol_202401201619](https://github.com/StudioPetrikas/Saulala/assets/59176246/849f02d1-8bdd-4b6f-b076-e3a49a202390)

**Comment by EaryChow** [comment]
> If you are unhappy with the 'default' processing

I don't understand why [your default](https://github.com/StudioPetrikas/Saulala/issues/11#issuecomment-1900793853) is different from [my default](https://github.com/StudioPetrikas/Saulala/issues/11#issuecomment-1901935303). Your default looks fine to me, while my default looks less chroma laden compared with yours. Our slider values are completely the same, how come we are having different result? I made sure to install the update before posting this issue, so it's probably not due to different versions.

**Comment by StudioPetrikas** [comment]
Yes, you're right. I cannot get the result you've posted. I suspected an error in 'Sauce' loading (since it feels like the reds are rotated towards orange like they are in "Saulala" sauce), but I cannot get that result with any sauce.

Are you on v0.2.10.0? (you can check in the About page)

I will look into this, thanks for the report.

**Comment by EaryChow** [comment]
> Are you on v0.2.10.0? (you can check in the About page)

Yes it's 0.2.10.0 from 2024.01.12

**Comment by StudioPetrikas** [comment]
Can you post an output jpg as a txt? The one you posted appears to be processed by GitHub. Thanks. 

The screenshot you're referring to is a screenshot of my AdobeRGB monitor, then converted to sRGB. I was just showing that the result to Darktable should be similar, so it will appear more chrome-laden. 

What is important to me that all apps on all phones produce identical image.

Here's the actual difference. My output from Saulala with AgX on the left, your attached picture on the right.

**Comment by EaryChow** [comment]
> Can you post an output jpg as a txt? The one you posted appears to be processed by GitHub. Thanks.

Sure, here it is:
[IMG_20230512_181009_Sol_202401201619.jpg.txt](https://github.com/StudioPetrikas/Saulala/files/14000346/IMG_20230512_181009_Sol_202401201619.jpg.txt)

**Comment by StudioPetrikas** [comment]
Just an update - we've pinpointed the problem and I'm now looking for a solution. 
At some point the raw file reader stopped (or never did) handed off sRGB array for processing and handed off camera primaries. 
On PC it was working as intended. 

Such a good catch by the way! There have been many points where I compared outputs and had perfect matches. At some point something happened out of my control and it never occurred to me that I need to periodically check if everything is still in order. 

Live and learn. Thanks for the bug reports, hopefully it's fixed in the next release.

**Comment by EaryChow** [comment]
> Thanks for the bug reports, hopefully it's fixed in the next release.

Tested the latest release, it's fixed!

BTW I managed to get an APK, but was unable to get it opened in Huawei, it seems Huawei's HarmonyOS is based on an older version of Android, this is probably why I still cannot install Saulala on Huawei. Judging by the other issue page here about older Android version, Saulala won't be supporting it. I am sad, but I guess sometimes that's how things go.
